src/main/java/org/folio/innreach/domain/service/impl/SystemUserAuthService.java

```diff
@@ -58,8 +54,7 @@ public void setupSystemUser() {
   }
 
   public String loginSystemUser(SystemUser systemUser) {
-    return executeWithinContext(contextBuilder.forSystemUser(systemUser), () -> {
-
+    try (var context = new FolioExecutionContextSetter(contextBuilder.forSystemUser(systemUser))) {
```

Comment:
It looks like sonar is complaining about this new try-with-resources block. Looks like it is considering the whole `try` block new code and it looks like the method has code coverage, but perhaps not the optional bit or the thrown exception. It would be nice if sonar was more specific here.